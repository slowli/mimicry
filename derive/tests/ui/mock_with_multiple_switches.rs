use mimicry_derive::CallReal;

/// Dummy struct to trick `CallReal` derive logic.
struct RealCallSwitch;

#[derive(CallReal)]
struct MyMock {
    #[mock(switch)]
    switch: RealCallSwitch,
    #[mock(switch)]
    another_switch: RealCallSwitch,
}

fn main() {}
