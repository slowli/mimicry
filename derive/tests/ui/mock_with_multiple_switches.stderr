error: Multiple `#[mock(switch)]` attrs; there should be no more than one
  --> tests/ui/mock_with_multiple_switches.rs:10:5
   |
10 | /     #[mock(switch)]
11 | |     another_switch: RealCallSwitch,
   | |__________________________________^
